src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.Helpers.cs

```diff
@@ -0,0 +1,366 @@
+            private const string FullNameForGenericDictionary = "System.Collections.Generic.Dictionary`2";
+
+            private const string LiteralFor_Bind = "Bind";
+            private const string LiteralFor_Get = "Get";
+            private const string LiteralFor_GetChildren = "GetChildren";
+            private const string LiteralFor_GetSection = "GetSection";
+            private const string LiteralFor_Key = "Key";
+            private const string LiteralFor_Parse = "Parse";
+            private const string LiteralFor_Value = "Value";
+
+            private const string LiteralFor_configuration = "configuration";
+            private const string LiteralFor_configurationSection = "configurationSection";
+            private const string LiteralFor_element = "element";
+            private const string LiteralFor_key = "key";
+            private const string LiteralFor_obj = "obj";
+            private const string LiteralFor_section = "section";
+            private const string LiteralFor_stringValue = "stringValue";
+
+            private const string NotSupportedReason_AbstractOrInterfaceNotSupported = "Abstract or interface types are not supported";
+            private const string NotSupportedReason_NeedPublicParameterlessCtor = "Only objects with public parameterless ctors are supported";
+            private const string NotSupportedReason_CollectionNotSupported = "The collection type is not supported";
+            private const string NotSupportedReason_CollectionKeyOrElementNotSupported = "The collection key or element type is not supported";
+            private const string NotSupportedReason_ValueTypesNotSupported = "Value types are not supported";
+            private const string NotSupportedReason_TypeNotDetectedAsInput = "Generator parser did not detect the type as input";
+
+            private const string MicrosoftExtensionsConfigurationBinderSourceGenerationName = "Microsoft.Extensions.Configuration.Binder.SourceGeneration";
+
+            private static DiagnosticDescriptor TypeNotSupported { get; } = new DiagnosticDescriptor(
+                id: "SYSLIB1091",
+                title: new LocalizableResourceString(nameof(SR.TypeNotSupportedTitle), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+                messageFormat: new LocalizableResourceString(nameof(SR.TypeNotSupportedMessageFormat), SR.ResourceManager, typeof(FxResources.Microsoft.Extensions.Configuration.Binder.SourceGeneration.SR)),
+                category: MicrosoftExtensionsConfigurationBinderSourceGenerationName,
+                defaultSeverity: DiagnosticSeverity.Info,
+                isEnabledByDefault: true);
+
+            // Unlike sourcegen warnings, exception messages should not be localized so we keep them in source.
+            private static class ExceptionMessages
```

Comment:
Nit: could this be moved to a standalone class? Or at least extract to another a Emitter.ExceptionMessages.cs source file?